### PR TITLE
Add a per-SteamClient unique identifier message to DebugLog messages

### PR DESCRIPTION
**Pull request by yaakov-h**
One approach to solving https://github.com/SteamRE/SteamKit/issues/503.

Thoughts?

Personally I don't like having to pass it everywhere - though I think that I would rather pass just a unique ID around than a whole `ILogger` object or similar.

I have considered storing it in an `AsyncLocal<T>` but that could get messy quite quickly.

(cc @JustArchi)

### CODE REVIEW

File: SteamKit2/SteamKit2/Networking/Steam3/NetFilterEncryption.cs [hunk 1]
```diff
@@ -12,12 +12,14 @@ namespace SteamKit2
     class NetFilterEncryption : INetFilterEncryption
     {
         readonly byte[] sessionKey;
+        readonly LoggerToken loggerToken;
 
-        public NetFilterEncryption( byte[] sessionKey )
+        public NetFilterEncryption( byte[] sessionKey ,LoggerToken loggerToken )
```
Comment (xPaw) [inline SteamKit2/SteamKit2/Networking/Steam3/NetFilterEncryption.cs]:
```suggestion
        public NetFilterEncryption( byte[] sessionKey, LoggerToken loggerToken )
```

**Comment by xPaw** [comment]
This is a breaking change for anyone that implements DebugLog, right?

Also, perhaps CMClient should accept a logger token guid so consumers can set one consistently if need be (same guid between restarts).

**Comment by yaakov-h** [comment]
As it is now, this is a breaking change.

If we want to keep compatibility, we could keep the old methods as overloads and wrap them, version `IDebugLogListener2`, even use DIM on .NET Core 3.0 (but keep a breaking change on other platforms).

**Comment by yaakov-h** [comment]
Abandoning in favour of #777.

**Pull request closed without merging**